diff --git a/dist/ts-morph-bootstrap.js b/dist/ts-morph-bootstrap.js
index 3df8fc8159b443510d13d0924a511d3b9d017696..a478c914301b2838a5edbcdf6ef52171a54ce172 100644
--- a/dist/ts-morph-bootstrap.js
+++ b/dist/ts-morph-bootstrap.js
@@ -262,6 +262,21 @@ class Project {
         const resolver = this.#getTsConfigResolver(tsConfigFilePath);
         return addSourceFilesForTsConfigResolverSync(this, resolver, resolver.getCompilerOptions());
     }
+    resolveTsConfig(tsConfigFilePath) {
+        const resolver = this.#getTsConfigResolver(tsConfigFilePath);
+        let compilerOptions;
+        return {
+            get compilerOptions() {
+                return (compilerOptions ??= resolver.getCompilerOptions());
+            },
+            get paths() {
+                return resolver.getPaths(this.compilerOptions);
+            },
+            get errors() {
+                return resolver.getErrors();
+            },
+        };
+    }
     #getTsConfigResolver(tsConfigFilePath) {
         const standardizedFilePath = this.#fileSystemWrapper.getStandardizedAbsolutePath(tsConfigFilePath);
         return new common.TsConfigResolver(this.#fileSystemWrapper, standardizedFilePath, this.compilerOptions.getEncoding());
diff --git a/lib/ts-morph-bootstrap.d.ts b/lib/ts-morph-bootstrap.d.ts
index c3ed153613a4cf3b0e5dc34d6846492055535c23..a3412e39a739309287f5eb695c7be174eefa69d5 100644
--- a/lib/ts-morph-bootstrap.d.ts
+++ b/lib/ts-morph-bootstrap.d.ts
@@ -304,6 +304,18 @@ export declare class Project {
      * @param tsConfigFilePath - File path to the tsconfig.json file.
      */
     addSourceFilesFromTsConfigSync(tsConfigFilePath: string): ts.SourceFile[];
+    /**
+     * Resolves the specified tsconfig.json file.
+     * @param tsConfigFilePath - File path to the tsconfig.json file.
+     */
+    resolveTsConfig(tsConfigFilePath: string): {
+        get compilerOptions(): ts.CompilerOptions;
+        get errors(): ts.Diagnostic[];
+        get paths(): {
+            filePaths: StandardizedFilePath[];
+            directoryPaths: StandardizedFilePath[];
+        };
+    };
     /**
      * Creates a source file at the specified file path with the specified text.
      *
